bash_completion

```diff
@@ -806,7 +806,7 @@ _comp_delimited()
     fi
     [[ $cur == *$delimiter* ]] && prefix=${cur%"$delimiter"*}$delimiter
 
-    if $deduplicate; then
+    if "$deduplicate"; then
```

Comment:
> I suppose there was a PR somewhere (which I've likely failed to follow up with) that would ensure a desired environment at start of a completion -- that would sound much preferable over this approach to me.

PR #739 and Issue #786 are related. PR #739 contains unrelated long discussions, but the current situation is summarized in the following two comments:

- https://github.com/scop/bash-completion/pull/739#issuecomment-1356376393
- https://github.com/scop/bash-completion/issues/786#issuecomment-1480199375

> To me, reading `if "$something"; then` raises the question "was this supposed to be `if [[ "$something" ]]; then`" (even though the quotes within the square brackets are superfluous) and thus makes the code heavier to read.

I actually prefer `[[ $something ]]` with `something` being an empty/non-empty variable. The reason that I decided to switch to this direction was item 2 in the review comment https://github.com/scop/bash-completion/pull/891#pullrequestreview-1314581077. Might I have misread the comment? Anyway, if you would like to switch to `[[ $something ]]`, I will adjust them.

Reply:
> I actually prefer `[[ $something ]]` with `something` being an empty/non-empty variable. The reason that I decided to switch to this direction was item 2 in the review comment [#891 (review)](https://github.com/scop/bash-completion/pull/891#pullrequestreview-1314581077). Might I have misread the comment?

You've got it right. My opinion is still that `if $something; then` reads cleaner than `if [[ $something ]]; then`. But per reasoning here, we can't have the former if we want to protect against funky `$IFS`. And I think `if "$something"; then` with the quotes is so weird that I prefer `if [[ $something ]]; then` over that.

> Anyway, if you would like to switch to `[[ $something ]]`, I will adjust them.

Yes, please. I'd also suggest using literal `set` as the value when the variable is set, instead of `true`, `1` or the like.

Reply:
> Yes, please. I'd also suggest using literal `set` as the value when the variable is set, instead of `true`, `1` or the like.

I reworked them so decided to submit them as another PR #931.